examples/wallet/program/src/main.rs

```diff
@@ -81,15 +76,11 @@ mod test {
   fn test_wallet() {
     let wallet = Wallet::new([0u8; 32]);
     let owner: Pubkey = [0u8; 32];
-    Wallet::spawn(owner.as_ptr());
+    Wallet::spawn(owner);
     let send_arguments = SendArguments {
       recipient: [0u8; 24],
       amount: 0,
     };
-    let serialized = to_vec(&send_arguments).unwrap();
-    wallet.send(serialized.as_ptr(), serialized.len());
-    unsafe { athexp_spawn(owner.as_ptr()) };
```

Comment:
Why remove this spawn call?

Reply:
Actually, it was not a mistake, I forgot :) I removed these direct calls to `athexp_` functions because they were wrong and did not compile. These functions don't take any arguments. You can use `cargo expand` to see them: `RUSTUP_TOOLCHAIN=athena cargo expand --target riscv32em-athena-zkvm-elf`:

```rust
pub unsafe extern "C" fn athexp_spawn() {
    athena_vm::program::Function::call_func(
        Wallet::spawn,
        &mut athena_vm::io::Io::default(),
    );
    syscall_halt(0);
}

pub unsafe extern "C" fn athexp_send() {
    athena_vm::program::Method::call_method(
        Wallet::send,
        &mut athena_vm::io::Io::default(),
    );
    syscall_halt(0);
}
```

Reply:
You can compile the test with: `RUSTUP_TOOLCHAIN=athena cargo t --no-run --target riscv32em-athena-zkvm-elf`

Reply:
okay. agree, it would be nice to have support for this eventually. should we add compilation of these tests to the CI pipeline?

Reply:
Added compiling these tests to the CI.